frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+  Typography,
+} from '@mui/material';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { post } from '../../lib/k8s/apiProxy';
+import Namespace from '../../lib/k8s/namespace';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton, AuthVisible } from '../common';
+
+export function CreateNamespaceButton() {
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [namespaceName, setNamespaceName] = useState('');
+  const [namespaceDialogOpen, setNamespaceDialogOpen] = useState(false);
+  const dispatchCreateEvent = useEventCallback(HeadlampEventType.CREATE_RESOURCE);
+
+  const dispatch = useDispatch();
+  const kubeNamespace = Namespace;
+
+  function toggleCreateNamespaceVisibility() {
+    setNamespaceDialogOpen(namespaceDialogOpen => !namespaceDialogOpen);
+  }
+
+  console.log('kubeNamespace', kubeNamespace);
+
+  function createNewNamespace() {
+    const newNamespaceData = {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+        name: namespaceName,
+      },
+    };
+
+    // need this for the item name?
+    const itemName = newNamespaceData.metadata.name;
+
+    // need a callback so just made one to use
+    const callback = (itemName: string) => {
+      console.log(`Callback called with ${itemName}`);
+    };
+
+    async function createNamesapce(newNamespaceData: any) {
+      try {
+        const response = await post('/api/v1/namespaces', newNamespaceData);
+        console.log('response', response);
+
+        toggleCreateNamespaceVisibility();
+        dispatch(
+          clusterAction(callback.bind(itemName), {
+            startMessage: t('Creating namespace {{ itemName }}…', { itemName }),
+            cancelledMessage: t('Cancelled creation of {{ itemName }}.', { itemName }),
+            successMessage: t('Created namesapce {{ itemName }}.', { itemName }),
+            errorMessage: t('Error creating namespace {{ itemName }}.', { itemName }),
+            cancelUrl: location.pathname,
+            // startUrl: item!.getListLink(),
+            // errorUrl: item!.getListLink(),
+            // ...options,
+          })
+        );
+
+        return response;
+      } catch (error) {
+        console.error('Error creating namespace', error);
+
+        toggleCreateNamespaceVisibility();
+        return error;
+      }
+    }
+
+    createNamesapce(newNamespaceData);
+  }
+
+  return (
+    <AuthVisible item={kubeNamespace} authVerb="create" namespace="default">
```

Comment:
Namespaces aren't namespaced. So the namespace argument is not needed here.
I am not sure the Namespace class can be used here like this either. Maybe AuthVisible needs changes to accommodate what we need.